src/app/containers/StoryPromo/index.jsx

```diff
@@ -1,18 +1,87 @@
 import React, { Fragment, useContext } from 'react';
+import moment from 'moment';
 import { shape } from 'prop-types';
+import MediaIndicator from '@bbc/psammead-media-indicator';
 import StoryPromoComponent, {
   Headline,
   Summary,
   Link,
 } from '@bbc/psammead-story-promo';
 import Timestamp from '@bbc/psammead-timestamp-container';
+import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
 import { storyItem } from '../../models/propTypes/storyItem';
 import StoryPromoFigure from './Figure';
 import { ServiceContext } from '../../contexts/ServiceContext';
 import deepGet from '../../lib/json/deepGet';
+import formatDuration from '../../lib/utilities/formatDuration';
+
+const buildMediaIndicator = (item, mediaTranslations) => {
+  const isMedia = deepGet(['cpsType'], item) === 'MAP';
+  // Only build a media indicator if this is a media item.
+  if (!isMedia) {
+    return null;
+  }
+
+  const type = deepGet(['media', 'format'], item);
+  // Always gets the first version. Smarter logic may be needed in the future.
+  const rawDuration = deepGet(['media', 'versions', 0, 'duration'], item);
+
+  if (rawDuration) {
+    const duration = moment.duration(rawDuration, 'seconds');
+    const durationString = formatDuration(duration);
+    const isoDuration = duration.toISOString();
+    return (
+      <MediaIndicator
+        duration={durationString}
+        datetime={isoDuration}
+        // TODO make offscreenText optional in psammead-media-indicator
+        offscreenText=""
+        type={type}
+      />
+    );
+  }
+
+  return <MediaIndicator offscreenText={mediaTranslations[type]} type={type} />;
+};
+
+const buildLinkContents = (item, mediaTranslations) => {
+  const isMedia = deepGet(['cpsType'], item) === 'MAP';
+
+  const headline = deepGet(['headlines', 'headline'], item);
+
+  if (!isMedia) {
+    return headline;
+  }
+
+  const type = deepGet(['media', 'format'], item);
+
+  // Always gets the first version. Smarter logic may be needed in the future.
+  const rawDuration = deepGet(['media', 'versions', 0, 'duration'], item);
+
+  // hilariously, this works. according to moment, null seconds == 0 seconds!
+  const duration = moment.duration(rawDuration, 'seconds');
+  const durationString = formatDuration(duration);
+  const isoDuration = duration.toISOString();
+
+  return (
+    // role="text" is required to correct a text splitting bug on iOS VoiceOver.
+    // eslint-disable-next-line jsx-a11y/aria-role
+    <span role="text">
+      <VisuallyHiddenText>{mediaTranslations[type]}, </VisuallyHiddenText>
+      <span>{headline}</span>
```

Comment:
If its not a media promo, do we not want to just return text here? one from @greenc05 but at a guess, we shouldn't wrap it in a span if we dont need to

Reply:
see https://github.com/bbc/simorgh/pull/1918/files/0b9a83cbcc14b637ca05edb0386c758f47a82b03#diff-5a217b9ae93005d18abe435f36b7ee9aR53

Reply:
Hahah whoops missed that 🙈